core/include/vecmem/containers/impl/device_vector.ipp

```diff
@@ -74,7 +81,7 @@ namespace vecmem {
    device_vector< TYPE >::at( size_type pos ) {
```

Comment:
Nitpick: isn't the usual assumption that `at()` methods always perform bounds checks, not only in DEBUG?

Reply:
This is something that we pondered about with @stephenswat as well early on.

In device code the only way in which a function like `at(...)` can fail is either through an `assert(...)`, or by performing some invalid memory operation explicitly. It does not have the option of throwing an exception, as those are "not a thing" in device code.

So... I definitely wanted to keep both the `at(...)` functions and the `[]` operators, to be as source-compatible with `std::vector` as possible. But I just couldn't come up with a way of properly mimicking the "safety" of `std::vector::at(...)` in `vecmem::device_vector::at(...)`. We should indeed reflect this in our documentation...

Reply:
Fair enough.